Support localization for parser messages

mdanish-kh opened the pull request:
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Are you working against an Issue?
	- Resolves #251

Reference implementation taken from the wiki: 

> To localize the keywords "Required,Default" and the error messages, add the the string expressions of the class SentenceBuilder to the resource files.

- https://github.com/commandlineparser/commandline/wiki/Help-Localization

- https://github.com/commandlineparser/commandline/blob/master/demo/ReadText.LocalizedDemo/LocalizableSentenceBuilder.cs
	
I modified the file to resolve any VS warnings

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-create/pull/457)

ryfu-msft commented on src/WingetCreateCLI/Properties/Resources.resx:
```diff
@@ -1121,4 +1121,67 @@
+  <data name="SentenceSequenceOutOfRangeErrorValue" xml:space="preserve">
+    <value>A sequence value not bound to option name is defined with fewer items than required.</value>
+  </data>
+  <data name="SentenceSetValueExceptionError" xml:space="preserve">
+    <value>Error setting value to option '{0}': {1}</value>
+  </data>
```
For all of these new localized strings that have a placeholder like `{0}` or`{1}`, you will need to add a comment explaining what they are placeholders for so that our localization team can perform the proper translations. 

Here is an example from some of our existing localized strings: https://github.com/microsoft/winget-create/blob/7d0f9458cf8551000d8ad530b7229b4192f54fda/src/WingetCreateCLI/Properties/Resources.resx#L231-L234

mdanish-kh replied:
I forgot to add them. 😄 Resolved

mdanish-kh commented:
@ryfu-msft Can we get a new release shipped out after this PR? #453 seems to be affecting a lot of CI runners in the wild. Fortunately, the regression didn't break any functionality but just that the CI runners exit with an exception and non-zero error code making the action show as "failed".

ryfu-msft commented:
> @ryfu-msft Can we get a new release shipped out after this PR? #453 seems to be affecting a lot of CI runners in the wild. Fortunately, the change didn't break any functionality but just that the CI runners exit with an exception and non-zero error code making the action show as "failed".

I will go ahead and cut a new release now before this PR gets checked in, since it will take some time for the strings to get localized. Thanks for calling that out 😄

ryfu-msft commented on src/WingetCreateCLI/Properties/Resources.resx:
```diff
@@ -1121,4 +1121,81 @@
+  <data name="SentenceMutuallyExclusiveSetErrors" xml:space="preserve">
+    <value>Options: {0} are not compatible with {1}.</value>
+    <comment>{0} - will be replaced with one or more provided command-line option(s)
+
```
nit: extra line not needed

ryfu-msft commented on src/WingetCreateCLI/Properties/Resources.resx:
```diff
@@ -1121,4 +1121,81 @@
+  <data name="SentenceSetValueExceptionError" xml:space="preserve">
+    <value>Error setting value to option '{0}': {1}</value>
+    <comment>{0} - will be replaced with the provided command-line option
+
```
nit: extra line not needed

ryfu-msft reviewed (changes requested):
just a couple nits for cleanup

mdanish-kh commented:
> just a couple nits for cleanup

I added the new lines that since I saw them being used in other comments. I went ahead and cleaned all of the other ones too

ryfu-msft commented:
/azp run
